### ISSUE TITLE
Error: `unable to parse link to update line number`

### ISSUE THREAD

**Issue by rgmz** [trufflesecurity/trufflehog#2052]
Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

### TruffleHog Version

3.61.0

### Description

Scanning a file with the path [`meta-xilinx/meta-xilinx-contrib/recipes-kernel/linux-firmware/linux-firmware_%.bbappend` (escaped as %25 in the url)](https://github.com/ibm-openbmc/openbmc/blob/286d45cc649a5b852a27b48e61b51bf75e9d1cee/meta-xilinx/meta-xilinx-contrib/recipes-kernel/linux-firmware/linux-firmware_%25.bbappend) produces the following error:

```
2023-10-30T09:13:31-04:00       error   trufflehog      unable to parse link to update line number
{
  "detector_worker_id": "e4wt4",
  "timeout": 10,
  "link": "https://github.com/ibm-openbmc/openbmc/blob/286d45cc649a5b852a27b48e61b51bf75e9d1cee/meta-xilinx/meta-xilinx-contrib/recipes-kernel/linux-firmware/linux-firmware_%.bbappend#L1",
  "error": "parse \"https://github.com/ibm-openbmc/openbmc/blob/286d45cc649a5b852a27b48e61b51bf75e9d1cee/meta-xilinx/meta-xilinx-contrib/recipes-kernel/linux-firmware/linux-firmware_%.bbappend\": invalid URL escape \"%.b\""
}
```

I'm not sure if there's anything that can be done to combat this. The error seems to be coming from `url.Parse`.

https://github.com/trufflesecurity/trufflehog/blob/2a66d4117a3f14b14f97e07668615bac905e4e5e/pkg/giturl/giturl.go#L155-L159

**Comment by rgmz** [comment]
This issue should be easily solved by replacing `%` with `%25` prior to calling url.Parse.

https://en.wikipedia.org/wiki/Percent-encoding

**Comment by rgmz** [comment]
Fixed by #2214.
